app/scripts/controllers/network/network-controller.test.js

```diff
@@ -4744,6 +5115,114 @@ describe('NetworkController', () => {
       });
     });
   });
+  describe('upsertNetworkConfiguration', () => {
```

Comment:
Nit: It would be great to add tests (for both the "new network" and "update" cases) that ensure existing network configurations aren't overwritten

Reply:
Maybe I'm misunderstanding the comment, but I believe these two cases are already covered: [new network](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-d4e78728faadb83c4ed8c0d5b5b580b026993a8988156b27bf7f41c4b5217af4R6280) & [update](https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-d4e78728faadb83c4ed8c0d5b5b580b026993a8988156b27bf7f41c4b5217af4R6314)

Reply:
Those links aren't correctly navigating to the referenced line for me, but it doesn't look like it's covered in this `describe` block.

There is a single test that adds a new network, and a single test that updates a pre-existing network. Neither one has additional network configurations, in addition to the one being added/updated.

Reply:
This test suite relies on an instance of the class that is instantiated at the top of the file. So there are existing networkConfigurations in the initial state: https://github.com/MetaMask/metamask-extension/pull/17421/files#diff-0aea0fd14a27750ddf078dd4ef363f4d10b91285954ad5e1488edcad1d5bd306R101.

They are certainly due a refactor, but I think that's out of scope for this PR

Reply:
sorry for the confusion here. Added some tests to address the cases you raise here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/aa91cce29a9c750b24c5d8d04a62298e2f604120